pkg/apply/taskrunner/task.go

```diff
@@ -178,6 +183,16 @@ func (w *WaitTask) sendTimeoutEvents(taskContext *TaskContext) {
 	}
 }
 
+// updateFailedReconciles updates the taskContext with resources that
+// are either stalled or still pending.
+func (w *WaitTask) updateFailedReconciles(taskContext *TaskContext) {
+	w.mu.RLock()
+	defer w.mu.RUnlock()
+
+	taskContext.AddFailedReconciles(w.pending)
```

Comment:
Why add pending as failed? 

sendTimeoutEvents sends these as ReconcileTimeout, which is not the same as Failed.

Reply:
So the idea here is that we want to capture the resources that failed to reconcile in the TaskContext. I was thinking that once a WaitTask completes, we don't really care whether a resource failed to reconcile due to a timeout or due to the controller reporting it as Failed/Stalled. In both cases, we have a resource where the controller wasn't able to reconcile it.
I agree though that for reporting this to users, it makes sense to distinguish between them. But for how we should handle the resource in later tasks, I don't think it matters.

Reply:
I guess the better question is: Why are we storing these in the task context if nothing is using `FailedReconciles()`?
Without usage, it's hard to know what should be in the set...

Reply:
Agree with that. Adding information to the TaskContext isn't strictly needed, so I have removed it from this PR.

Reply:
Removed this from the PR. We can add it back when it is needed.